### ISSUE TITLE
drop = FALSE does not work in scale_*_manual

### ISSUE THREAD

**Issue by cpauvert** [thomasp85/ggraph#164]
Hi,
Thank you very much for this package! However, I was puzzled to see that the option `drop = FALSE` was not passed along when provided through `...` in `scale_color_manual` for instance.

```
library(igraph)
gr <- make_graph('bull')
#Adding a categorical value
V(gr)$Cat<-c("A","A","A","B","B")
library(ggraph) # v1.0.2     and ggplot2 v3.0.0
layout <- create_layout(gr, layout = 'igraph', algorithm = 'kk')
ggraph(layout)+geom_node_point(aes(color=Cat))+geom_edge_link()+
  coord_fixed()+theme_void()+
  # Trying to map all possible values
  scale_color_manual(values = c("A"="red","B"="blue","C"="black"),drop=FALSE)
```

Only the red and blue nodes are shown in the plot, which is expected. But the legend should mention the "C" category and the black label.

Do you have any ideas?
Best, 
Charlie

**Comment by malcolmbarrett** [comment]
This is just because `Cat` only has levels "A" and "B". The same behavior occurs in ggplot.

``` r
df <- data.frame(x = 1:3, y  = 1:3, z = factor(c("A", "B", "B")))

ggplot(df, aes(x, y, col = z)) + 
  geom_point() +
  scale_color_manual(values = c("A" = "red", "B" = "blue", "C" = "black"), drop=FALSE)
```

![](https://i.imgur.com/JzFDK70.png)

So you just need to tell it the levels.

``` r
layout$Cat <- factor(layout$Cat, levels = c("A", "B", "C"))

ggraph(layout) +
  geom_node_point(aes(color = Cat)) + 
  geom_edge_link() +
  coord_fixed()+
  theme_void()+
  scale_color_manual(values = c("A" = "red","B" = "blue","C" = "black"), drop=FALSE)
```

![](https://i.imgur.com/KVWkrdh.png)

**Comment by cpauvert** [comment]
Hi!
Thank you very much @malcombarrett ! I overlooked factors indeed.
Best,
Charlie

**Comment by Artur-man** [comment]
I think this doesnt work anymore because the same scripts above is giving empty point in the legend

<img width="1009" alt="Screenshot 2024-05-21 at 17 08 40" src="https://github.com/thomasp85/ggraph/assets/39335576/73cf8643-41bc-47d8-aa24-ea48e5bd087a">

**Comment by goesjon** [comment]
Same problem here but haven't found a solution yet. If anyone has a solution, please let me know.

**Comment by sporella** [comment]
Same problem with ggplot2 version 3.5.0. My "solution" was return to version 3.4.0

**Comment by sporella** [comment]
Its a change included in ggplot2 version 3.5.0  [more info here](https://github.com/tidyverse/ggplot2/issues/5728)
